index.js

```diff
@@ -35,14 +35,26 @@ module.exports = function (babel) {
     return result;
   }
 
-  function parseScopeObject(buildError, name, node) {
-    if (node.type !== 'ObjectExpression') {
+  function parseScope(buildError, name, node) {
+    let body;
+
+    if (node.type === 'ArrowFunctionExpression') {
```

Comment:
IMHO, we should handle both arrow and non-arrow here. Either way, we will need to forbid doing anything other than returning, but as an authoring format this looks much better to me than the current suggestion:

```js
import { precompileTemplate } from '@ember/template-compilation';
const compiled = precompileTemplate('some template here', {
  scope() {
    return { foo, bar };
  }
});
```

And I just don't see any reason for the uncanny valley (why only an arrow function? why does it have to be a non-block arrow function? ugh those paren around returning an object are fugly!?!?).

Reply:
The main reason for it is simplicity, while this is an authoring format, it's also pretty much guaranteed not to be the final authoring format. I think handling the different permutations of types of object expressions opens us up to more churn and maintenance overall.